test/uk/gov/hmrc/economiccrimelevyregistration/controllers/deregister/DeregistrationRequestedControllerSpec.scala

```diff
@@ -123,20 +122,20 @@ class DeregistrationRequestedControllerSpec extends SpecBase {
             .thenReturn(EitherT[Future, DataRetrievalError, Unit](Future.successful(Right(()))))
 
           val result: Future[Result] = controller.onPageLoad()(
-            fakeRequest
+            fakeRequest.withSession(SessionKeys.FirstContactEmail -> email)
           )
 
           status(result) shouldBe INTERNAL_SERVER_ERROR
         }
     }
 
-    "return InternalServerError when the email address is not present in the deregistration" in forAll {
+    "return InternalServerError when the email address is not present on the session" in forAll {
       (
         email: String,
         deregistration: Deregistration
       ) =>
         val updatedDeregistration =
-          deregistration.copy(contactDetails = deregistration.contactDetails.copy(emailAddress = None))
+          deregistration.copy(contactDetails = deregistration.contactDetails.copy(emailAddress = Some(email)))
```

Comment:
I don't believe this is now necessary, as we are not pulling the deregistration from the retrieval action, so it can be removed from the Test Context and all the tests here.

Reply:
Updated tests.